### PR TITLE
[Layout foundations] Remove default spacing from AlphaStack

### LINKED ISSUE #8369: [Layout foundations] Remove default spacing from `AlphaStack`
As per https://github.com/Shopify/polaris/discussions/8329
- [ ] Remove default `gap` of `4` and replace with `0`
- [ ] Update usages in Polaris as required

### PR DESCRIPTION
**Pull request by chazdean**
### WHY are these changes introduced?

Fixes #8369 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page, AlphaStack} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
         <AlphaStack>
            <Badge>Paid</Badge>
            <Badge>Processing</Badge>
            <Badge>Fulfilled</Badge>
            <Badge>Completed</Badge>
        </AlphaStack>
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/AlphaStack/AlphaStack.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 .AlphaStack {
   // stylelint-disable -- Polaris component custom properties
-  --pc-stack-gap-xs: var(--p-space-4);
+  --pc-stack-gap-xs: none;
```
Comment (aveline) [inline polaris-react/src/components/AlphaStack/AlphaStack.scss]: `gap: none` isn't valid CSS 
```suggestion
  --pc-stack-gap-xs: 0;
```

**Review by kyledurand — COMMENTED** [review]
You'll have to remove the [`@default` ](https://github.com/Shopify/polaris/pull/8383/files#diff-8a63ca0b0aaf5fea56c27bda3b9ac116ab5cbed9faffa7879629a0bccb2c6eb4R34)and run `yarn get-props` for the styleguide as well

**Comment by chazdean** [comment]
> You'll have to remove the [`@default` ](https://github.com/Shopify/polaris/pull/8383/files#diff-8a63ca0b0aaf5fea56c27bda3b9ac116ab5cbed9faffa7879629a0bccb2c6eb4R34)and run `yarn get-props` for the styleguide as well

@kyledurand do you think it should it be removed completely or changed to `@default 0` ?

File: .changeset/empty-dancers-think.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
```
Comment (laurkim) [inline .changeset/empty-dancers-think.md]:
```suggestion
'@shopify/polaris': minor
```

**Comment by laurkim** [comment]
> > You'll have to remove the [`@default` ](https://github.com/Shopify/polaris/pull/8383/files#diff-8a63ca0b0aaf5fea56c27bda3b9ac116ab5cbed9faffa7879629a0bccb2c6eb4R34)and run `yarn get-props` for the styleguide as well
> 
> @kyledurand do you think it should it be removed completely or changed to `@default 0` ?

I think we should remove the default completely

File: polaris-react/src/components/AlphaStack/AlphaStack.tsx [hunk 3]
```diff
@@ -30,9 +30,7 @@ export interface AlphaStackProps extends React.AriaAttributes {
    * @default false
    */
   fullWidth?: boolean;
-  /** The spacing between children
-   * @default '4'
-   */
+  /** The spacing between children*/
```
Comment (laurkim) [inline polaris-react/src/components/AlphaStack/AlphaStack.tsx]:
```suggestion
  /** The spacing between children */
```

**Review by laurkim — APPROVED** [review]
Nice 💯

**Pull request merged**